qrexec/tools/qrexec_legacy_convert.py

```diff
@@ -0,0 +1,240 @@
+            rules_to_save[filename] = []
+            for rule in missing:
+                if rule.rule.action == 'allow':
+                    # tool does not support allow-rules
+                    rules_to_save[CONFIG_FILE].append(rule)
+                    continue
+                rules_to_save[filename].append(rule)
+
+        elif service == 'qubes.Filecopy':
+            filename = '50-config-filecopy'
+            rules_to_save[filename] = missing
+            continue
+
+        elif service == 'qubes.UpdatesProxy':
+            filename = '50-config-updates'
+            rules_to_save[filename] = missing
+            continue
+
+        elif service in ('qubes.OpenInVM', 'qubes.OpenURL'):
+            filename = '50-config-openinvm' \
+                if service == 'qubes.OpenInVM' else '50-config-openurl'
+            rules_to_save[filename] = []
+            for rule in missing:
+                if rule.rule.target != '@dispvm':
+                    # tool does not support cases other than dispvm-related
+                    rules_to_save[CONFIG_FILE].append(rule)
+                    continue
+                if str(rule.rule.action) == 'ask':
+                    target = str(rule.rule.action.default_target)
+                elif str(rule.rule.action) == 'allow':
+                    target = str(rule.rule.action.target)
+                else:
+                    target = '@dispvm'
+                if '@dispvm' not in target:
+                    rules_to_save[CONFIG_FILE].append(rule)
+                    continue
+                rules_to_save[filename].append(rule)
+            rules_to_save[filename] = missing
+
+        elif service == 'qubes.Gpg':
+            filename = '50-config-splitgpg'
+            rules_to_save[filename] = []
+            for rule in missing:
+                if rule.rule.source == '@anyvm':
+                    if rule.rule.target.type == 'keyword':
+                        # not supported main rule
+                        rules_to_save[CONFIG_FILE].append(rule)
+                    else:
+                        rules_to_save[filename].append(rule)
+                else:
+                    rules_to_save[filename].append(rule)
+        elif service in ('qubes.InputKeyboard',
+                             'qubes.InputMouse',
+                             'qubes.InputTablet'):
+            filename = '50-config-input'
+            if filename not in rules_to_save:
+                rules_to_save[filename] = []
+
+            rule = missing.pop()
+
+            if rule.rule.target != '@adminvm':
+                rules_to_save[CONFIG_FILE].append(rule)
+            elif rule.rule.action.user:
+                rules_to_save[CONFIG_FILE].append(rule)
+            else:
+                rules_to_save[filename].append(rule)
+            if missing:
+                rules_to_save[CONFIG_FILE].extend(missing)
+
+        elif service in ('u2f.Authenticate', 'u2f.Register',
```

Comment:
This is not specific enough - it puts rules for specific U2F keys in the policy file too, but they aren't currently handled correctly by the global config tool: https://github.com/QubesOS/qubes-issues/issues/8525

Reply:
this probably means those rules with a specified argument should go into their appropriate file - `60-registered-arguments.policy`